packages/content-conformance/src/runner.ts

```diff
@@ -50,14 +59,44 @@ export class ContentConformanceRunner {
     })
   }
 
+  getStatisticsStatus(statistics: Statistics, warnThreshold?: number) {
+    if (
+      statistics.fatal > 0 ||
+      (warnThreshold && statistics.warn >= warnThreshold)
+    ) {
+      return RunnerStatus.FAILURE
+    }
+
+    return RunnerStatus.SUCCESS
+  }
+
+  /**
+   * TODO: Determine best ways to surface warnThreshold to user & default warnThreshold
+   */
   async run() {
-    return this.engine?.execute()
+    if (this.status === RunnerStatus.RUNNING) return null
+
+    // @ts-expect-error -- need to sort out VFile types here as well
+    const _statistics = statistics(this.engine?.files)
+
+    this.status = RunnerStatus.RUNNING
+    try {
+      await this.engine?.execute()
+    } catch {
+      this.status = RunnerStatus.FAILURE
+    }
+
+    /**
+     * check vFile-statistics for fatal messages, optionally pass in a warning count threshold
+     * getStatisticsStatus(statistics, warnThreshold)
+     */
+    this.status = this.getStatisticsStatus(_statistics)
```

Comment:
This would override the assignment in the catch block above, can we move this to right after L84 inside the try block?